src/webui/frontend/src/components/RuleEditors/RawEditor.tsx

```diff
@@ -7,23 +7,46 @@ import {
   CheckCircleOutlined,
 } from '@ant-design/icons';
 
-import * as React from 'react';
-import {connect} from 'react-redux';
-import {bindActionCreators, Dispatch} from 'redux';
+  import * as React from 'react';
+  import {connect} from 'react-redux';
+  import {bindActionCreators, Dispatch} from 'redux';
 
-import {getRules} from '../../reducers/rules';
-import {updateRuleBody, saveRule, deleteRule} from '../../actions/rules';
+  import {getRules} from '../../reducers/rules';
+  import {updateRuleBody, saveRule, deleteRule} from '../../actions/rules';
 
-import {State, SnowAlertRulesState} from '../../reducers/types';
-import sqlFormatter from 'snowsql-formatter';
+  import {State, SnowAlertRulesState} from '../../reducers/types';
+  import sqlFormatter from 'snowsql-formatter';
+  import { useRef, useEffect, useState } from "react";
 
+import { EditorView, ViewUpdate, highlightSpecialChars, drawSelection, highlightActiveLine, keymap} from "@codemirror/view";
+import { EditorState, Prec} from "@codemirror/state";
```

Comment:
There should be a separate wrapper file for CM6 code editor and all the codemirror 6 imports should happen in that file. Otherwise, it adds a lot of noise to the main code.